package/src/components/Channel/hooks/useMessageListPagination.tsx

```diff
@@ -0,0 +1,246 @@
+import { useRef } from 'react';
+
+import debounce from 'lodash/debounce';
+import { Channel, ChannelState } from 'stream-chat';
+
+import { useChannelMessageDataState } from './useChannelDataState';
+
+import { ChannelContextValue } from '../../../contexts/channelContext/ChannelContext';
+import { DefaultStreamChatGenerics } from '../../../types/types';
+
+const defaultDebounceInterval = 500;
+const debounceOptions = {
+  leading: true,
+  trailing: true,
+};
+
+/**
+ * The useMessageListPagination hook handles pagination for the message list.
+ * It provides functionality to load more messages, load more recent messages, load latest messages, and load channel around a specific message.
+ *
+ * @param channel The channel object for which the message list pagination is being handled.
+ */
+export const useMessageListPagination = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>({
+  channel,
+}: {
+  channel: Channel<StreamChatGenerics>;
```

Comment:
Total nitpick and definitely not something that blocks this PR, but since all of these message related hooks pretty much imply that they need to be used within a channel (we don't support message lists outside of one anyhow) perhaps we can make the `channel` optional as an argument and if it doesn't exist use the one from `useChannelContext`. Not really important, just a thought.

Reply:
Makes sense. I will do it.